File: solutions/go/bird-watcher/1/bird_watcher.go
package birdwatcher

// TotalBirdCount return the total bird count by summing
// the individual day's counts.
func TotalBirdCount(birdsPerDay []int) int {
    var totalBirds int
	for i := 0; i < len(birdsPerDay); i++ {
        totalBirds += birdsPerDay[i]
    }
    return totalBirds
}

// BirdsInWeek returns the total bird count by summing
// only the items belonging to the given week.
func BirdsInWeek(birdsPerDay []int, week int) int {
	firstIndex := (week * 7) - 7
    lastIndex := (week * 7) -1
	var totalBirds int
    for i := firstIndex; i <= lastIndex; i++ {
        totalBirds += birdsPerDay[i]
    } 
    return totalBirds
}

// FixBirdCountLog returns the bird counts after correcting
// the bird counts for alternate days.
func FixBirdCountLog(birdsPerDay []int) []int {
	for i := 0; i < len(birdsPerDay); i++ {
        if i % 2 == 0 {
            birdsPerDay[i]++
        }
    }
    return birdsPerDay
}

File: solutions/javascript/bird-watcher/1/bird-watcher.js
// @ts-check
//
// The line above enables type checking for this file. Various IDEs interpret
// the @ts-check directive. It will give you helpful autocompletion when
// implementing this exercise.

/**
 * Calculates the total bird count.
 *
 * @param {number[]} birdsPerDay
 * @returns {number} total bird count
 */
export function totalBirdCount(birdsPerDay) {
  var count = 0;
  for (let day = 0; day < birdsPerDay.length; day++) {
    count += birdsPerDay[day];
  }
  return count;
}

/**
 * Calculates the total number of birds seen in a specific week.
 *
 * @param {number[]} birdsPerDay
 * @param {number} week
 * @returns {number} birds counted in the given week
 */
export function birdsInWeek(birdsPerDay, week) {
  var count = 0;
  for (let day = 7 * (week-1); day < week * 7; day++) {
    count += birdsPerDay[day];
  }
  return count;
}

/**
 * Fixes the counting mistake by increasing the bird count
 * by one for every second day.
 *
 * @param {number[]} birdsPerDay
 * @returns {number[]} corrected bird count data
 */
export function fixBirdCountLog(birdsPerDay) {
  for (let day = 0; day < birdsPerDay.length; day++) {
    if (day % 2 == 0) {
      birdsPerDay[day] += 1;
    }
  }
  return birdsPerDay;
}
